mungegithub/mungers/matchers/comment/finder.go

```diff
@@ -0,0 +1,104 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package comment
+
+import "github.com/google/go-github/github"
+
+// Filter let's you retrieve your filtered comments
+type Filter struct {
+	comments []*github.IssueComment
+	matcher  Matcher
+}
+
+// Iterator over the list of filtered comments
+type Iterator struct {
+	ascending bool
+	current   int
+	filter    *Filter
+}
+
+func (it *Iterator) increment() int {
+	if it.ascending {
+		return 1
+	}
+	return -1
+}
+
+// Next looks for the next item. Returns true if there is one, false otherwise.
+func (it *Iterator) Next() bool {
+	for it.current += it.increment(); it.current >= 0 && it.current < len(it.filter.comments); it.current += it.increment() {
```

Comment:
drive by as I'm seeing this in my email: it's a little weird to see an
iterator in go.

On Tue, Aug 16, 2016 at 1:53 PM, Phillip Wittrock <notifications@github.com>
wrote:

> In mungegithub/mungers/matchers/comment/finder.go
> <https://github.com/kubernetes/contrib/pull/1530#discussion_r75017914>:
>
> > +type Iterator struct {
> > +	ascending bool
> > +	current   int
> > +	filter    *Filter
> > +}
> > +
> > +func (it *Iterator) increment() int {
> > +	if it.ascending {
> > +		return 1
> > +	}
> > +	return -1
> > +}
> > +
> > +// Next looks for the next item. Returns true if there is one, false otherwise.
> > +func (it *Iterator) Next() bool {
> > +	for it.current += it.increment(); it.current >= 0 && it.current < len(it.filter.comments); it.current += it.increment() {
>
> Consider
>
> for ; it.current >= 0 && it.current < len(it.filter.comments);
> it.increment() {
>
> }
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/kubernetes/contrib/pull/1530/files/9ab6ca864d0d7dec72e30d34333cc9ca8df40636#r75017914>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAnglpUe3ihMqdiqlYnedt-5TNcpZdlQks5qgiNVgaJpZM4JjXFP>
> .
>